Acumatica.RESTClient/BaseApi/ODataAPI.cs

```diff
@@ -12,136 +12,68 @@
 namespace Acumatica.RESTClient.Api
 {
 
-    public class ODataAPI
+    public class ODataAPI : BaseApi
     {
-        protected readonly string Username;
-        protected readonly string Password;
-        protected string BasePath;
+      
         protected ODataVersion Version;
         protected string Tenant;
         protected Token Token;
         
-        //for oauth initialization
-        public ODataAPI(Configuration configuration, ODataVersion version, string tenant=null) 
+        public ODataAPI(Configuration configuration, ODataVersion version, string tenant=null): base(configuration)
         {
-            Username = configuration.Username;
-            Password = configuration.Password;
-            BasePath = configuration.BasePath;
             Version = version;
             Tenant = tenant;
             Token = configuration.Token;
-            
-
-        }
-        
-        //for basic authentication
-        public ODataAPI(string username, string password, string basePath, ODataVersion version, string tenant = null)
-        {
-            Username = username;
-            Password = password;
-            BasePath = basePath;
-            Version = version;
-            Tenant = tenant;
         }
       
-        public string Get(string resource = null, string select = null, string filter = null, string expand = null, string custom = null, int? skip = null, int? top = null)
+       
+        public ApiResponse<string> Get(string resource = null, string select = null, string filter = null, string expand = null, string custom = null, int? skip = null, int? top = null)
         {
-            RestRequest request;
-            RestResponse response;
-            var path = ConfigurePath(Version.ToString(), Tenant);
-            RestClient client = new RestClient(path);
-            //Oauth authentication
-            if (Token != null)
-            {
-                request = GetOauthAuthentication(client);
-            }   
+            var path = ConfigureResourcePath(ConfigurePath(), resource);
 
             //Basic authentication
-            else
+            if (Token == null)
             {
-                request = GetBasicAuthentication(client, resource);                
+                BasicAuthentication(this.Configuration.ApiClient.RestClient);
```

Comment:
It's a good idea to check whether the Login and Password are filled in the provided configuration in that case. 
If neither Token nor Login/Password pair are provided, I think we need to throw an Exception with a clear error message.